### ISSUE TITLE
Empty EVM bytecode after upload

### ISSUE THREAD

**Issue by ducphamle2** [oraichain/wasmd#11]
## Context
When running the e2e test, I encountered an EVM bug of not being able to query or execute any EVM contracts after uploading. The request always returned an empty response.

## To Reproduce

1. Upload an EVM Solidity contract. Eg: ERC20 contraxt
2. Invoke a simple contract query. Eg: a decimals() query

## Expected behavior

A `decimals()` query should return the contract's decimals()

## Screenshots

![image](https://github.com/user-attachments/assets/2b90ca8c-b3ac-42d7-a605-b4af76bff91c)

**Comment by ducphamle2** [comment]
After a day of debugging, I found the root cause of the bug. The [Ethermint](https://github.com/oraichain/ethermint-cosmos-0.50) repo was querying an EVM contract's bytecodes from an EVM account casted from the AccountI of `auth` module.

Below is the `GetCode` logic to get the EVM bytecodes before querying or executing the contract:
<img width="619" alt="image" src="https://github.com/user-attachments/assets/a4d6c531-85dc-4877-9d64-da91e95dcb51">

**Comment by ducphamle2** [comment]
If the `stateObject` is not available -> we will query the contract's account and use its bytecodes:

<img width="632" alt="image" src="https://github.com/user-attachments/assets/555d5a88-aec4-4cc0-97fc-4aba59e66c3b">

However, the `SetAccount` method, doesn't store any contract's code hash because by default, every `BaseAccount` is not an `EthAccount`:

<img width="771" alt="image" src="https://github.com/user-attachments/assets/88730f5b-385e-4c8c-b388-5c47e4f27939">

Note that the `CodeHash` field is the EVM contract bytecodes' key. If it isn't stored anywhere -> can't retrieve the bytecodes

**Comment by ducphamle2** [comment]
A simple fix is to add the following logic to `SetAccount`:

```go
if !ok && account.IsContract() {
		if baseAcct, isBaseAccount := acct.(*authtypes.BaseAccount); isBaseAccount {
			acct = &ethermint.EthAccount{
				BaseAccount: baseAcct,
				CodeHash:    codeHash.Hex(),
			}
		} else {
			return errorsmod.Wrapf(types.ErrInvalidAccount, "type %T, address %s", acct, addr)
		}
	}
```

It will check if the account is a contract. If it is, then it will cast the `BaseAccount` to `EthAccount` and store it

**Comment by ducphamle2** [comment]
This bug is resolved in #13. The actual implementation is in https://github.com/oraichain/ethermint-cosmos-0.50/commit/3c8d318e25d69eae4d8a567a988a4b9a7f4d1443
